Revit_Core_Engine/Query/Bounds.cs

```diff
@@ -45,16 +46,44 @@ public static BoundingBoxXYZ Bounds(this IEnumerable<Solid> solids, Transform tr
             if (solids == null || !solids.Any())
                 return null;
 
-            Solid union = solids.First();
-            foreach (Solid solid in solids.Skip(1))
+            double minX = double.MaxValue;
+            double minY = double.MaxValue;
+            double minZ = double.MaxValue;
+            double maxX = double.MinValue;
+            double maxY = double.MinValue;
+            double maxZ = double.MinValue;
+
+            foreach (Solid solid in solids)
             {
-                union = BooleanOperationsUtils.ExecuteBooleanOperation(union, solid, BooleanOperationsType.Union);
+                BoundingBoxXYZ solidBBox = new BoundingBoxXYZ();
+
+                if (transform != null)
+                {
+                    Solid transSolid = SolidUtils.CreateTransformed(solid, transform);
+                    solidBBox = transSolid.GetBoundingBox();
+                }
+                else
+                {
+                    solidBBox = solid.GetBoundingBox();
+                }
```

Comment:
This could be done in 4 lines:
```suggestion
                if (transform != null)
                    solid = SolidUtils.CreateTransformed(solid, transform);
                
                BoundingBoxXYZ solidBBox = solid.GetBoundingBox();
```